### PR TITLE
Branches - Michaela

### PR DESCRIPTION
**Pull request by michaela260**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The Model is responsible for much of the "thinking" behind the program. It handles the logic and communicates with the database to get the actual information necessary.
Describe in your own words what the Controller is doing in Rails | The controller is telling the other pieces what to do. It gets a request from the user, tells the Model and View what to do, and then sends a response back to the user.
Describe in your own words what the View is doing in Rails | The view is what the users actually see when they look at the program. It includes the HTML and design of the application.
Describe an edge-case controller test you wrote | An edge case I wrote for the "update" method was that it would redirect to the root page if given an invalid id (i.e. a task id number that doesn't exist) to update.
What is the purpose of using strong params? (i.e. the params method in the controller) | The purpose of strong params is for security, to make sure that someone else can't edit your forms from another location, I think. They help solve this problem by utilizing private methods.
How are Rails migrations related to Rails models? | Rails migrations are a way to update the database schema, or the way that the database is organized. The model in rails is what communicates with the database, so if the schema is changed with a migration it affects the model.
Describe one area of Rails that are still unclear on | I am still a bit unclear about the router and differences between the router and controller.

### CODE REVIEW

**Comment by tildeee** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | x
Answered comprehension questions | x
Successfully handles: Index, Show | x
Index & Show tests pass | x
Successfully handles: New, Create | x
New & Create tests pass |x
Successfully handles: Edit, Update | x
Tests for Edit & Update test valid & invalid task ids | x
Successfully handles: Destroy, Task Complete | x
Tests for Destroy & Task Complete include tests for valid and invalid task ids | x
Routes follow RESTful conventions | x
Uses named routes (like `_path`) | x
**Overall** |



Well done on this project, Michaela! The app code and test code look great.


Your code has a very classic style-- it looks like exactly what I'm looking for! Also, your tests are particularly readable and thorough and detailed. Honestly, your tests look great and you have all the cases I was looking for.


Well done on this project! Keep up the hard work!

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
+class TasksController < ApplicationController
+  
+  def index
+    @tasks = Task.all.order(:id)
```
Comment (tildeee) [inline app/controllers/tasks_controller.rb:4]: Nice detail on ordering/sorting!

File: test/controllers/tasks_controller_test.rb [hunk 2]
```diff
@@ -71,56 +70,167 @@
...
   # Complete for Wave 4
   describe "toggle_complete" do
     # Your tests go here
+    it "succesfully updates the completion date to not be nil if it was previously nil and redirects to index page" do
+      new_task_1 = Task.create(name: "Toggle test task!", description: "Testing the toggle task")
+      expect(new_task_1.completion_date).must_be_nil
+      test_id = new_task_1.id
+      
+      patch toggle_path(test_id)
+      new_task_1.reload
```
Comment (tildeee) [inline test/controllers/tasks_controller_test.rb:206]: Nice! `.reload` is a clean way to reload the data, and it looks really clean too! Keep doing this whenever this makes sense!
